src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,116 @@
+package duke.task;
+
+/**
+ * The `Task` class represents a basic task with a description and a completion status.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a new task with the given description, initially marked as not done.
+     * @param description The description of the task.
+     */
+    public Task (String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the type of the task. Subclasses should override this method.
+     * @return An empty string for the base class.
+     */
+    public String getTaskType() {
+        return ""; // Return an empty string for the base class
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as undone.
+     */
+    public void markNotDone() {
+        this.isDone = false;
+    }
+
+    /**
+     * Checks if the task is marked as done.
+     * @return `true` if the task is done, otherwise `false`.
+     */
+    public boolean getIsDone() {
+        return this.isDone;
+    }
+
+    /**
+     * Returns a string representation of the task's completion status.
+     * @return "X" if the task is done, or a space character if it's not done.
+     */
+    public String taskStatus() {
+        return isDone ? "X" : " ";
+    }
+
+    /**
+     * Retrieves the description of a task.
+     * @return The description of the task as a string.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Converts the task to a string representation for saving to a file.
+     * @return A formatted string representing the task's data.
+     */
+    public String toFileString() {
+        char doneStatus = isDone ? '1' : '0';
+        return doneStatus + " | " + description;
+    }
+
+    /**
+     * Converts the task to a string representation for displaying to the user.
+     * @return A formatted string representing the task's type, status, and description.
+     */
+    @Override
+    public String toString() {
+        return getTaskType() + " [" + taskStatus() + "] " + description;
+    }
+
+    /**
+     * Creates a task object from a data string
```

Comment:
An empty line between the description and parameter section of JavaDoc comments seems to be the practice recommended by our coding standard? I noticed this issue in several other places in the code base as well.